### PR TITLE
Clean up warnings to make it easier to see actual issues

### PR DESCRIPTION
**Pull request by dsargent**
I verified that Codefresh detects an intentional test failure that I injected, and that it runs all tests, of which there are 25.

I also cleaned up the logging noise so it's easier to see what's actually going on with the tests.

### CODE REVIEW

File: examples/predictors/lstm/xprize_predictor.py [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 import os
 
+# Suppress noisy Tensorflow debug logging
+os.environ['TF_CPP_MIN_LOG_LEVEL'] = '3'
```
Comment (dsargent) [inline examples/predictors/lstm/xprize_predictor.py:6]: This is a bit drastic as it silences all the messages from Tensorflow, but it's not like anyone ever reads them. One of the more annoying ones is due to the lack of CUDA, even on non-GPU-enabled hardware. I learned there is a "lean" Tensorflow package, `tensorflow-cpu`, that doesn't include support for GPUs, which we could look into using some day.

File: examples/predictors/lstm/xprize_predictor.py [hunk 2]
```diff
@@ -212,7 +215,8 @@ def _load_original_data(self, data_url, cutoff_date=None):
         latest_df = pd.read_csv(data_url,
                                 parse_dates=['Date'],
                                 encoding="ISO-8859-1",
-                                error_bad_lines=False)
+                                error_bad_lines=False,
+                                low_memory=False)
```
Comment (dsargent) [inline examples/predictors/lstm/xprize_predictor.py:219]: Silencing a Pandas warning

File: validation/scenario_generator.py [hunk 3]
```diff
@@ -52,7 +52,7 @@ def generate_scenario(start_date_str, end_date_str, raw_df, countries=None, scen
         if start_date < INCEPTION_DATE:
             raise ValueError(f"start_date {start_date} must be on or after inception date {INCEPTION_DATE}")
 
-    ips_df = raw_df[ID_COLS + NPI_COLUMNS]
+    ips_df = raw_df[ID_COLS + NPI_COLUMNS].copy()
```
Comment (dsargent) [inline validation/scenario_generator.py:55]: Silencing another pandas warning. It looks like the Dataframes are small here so making a deep copy shouldn't be expensive.

File: examples/predictors/lstm/xprize_predictor.py (hunk 2 above)
Comment (ofrancon) [reply examples/predictors/lstm/xprize_predictor.py:219]: I've fixed it in https://github.com/leaf-ai/covid-xprize/pull/93
The `RegionName` and `RegionCode` columns contain a string or NaN, preventing Pandas from figuring out the type of the columns. Fixing the type to `str` is enough.

File: validation/scenario_generator.py (hunk 3 above)
Comment (ofrancon) [reply validation/scenario_generator.py:55]: Same as above

File: examples/predictors/lstm/xprize_predictor.py (hunk 1 above)
Comment (ofrancon) [reply examples/predictors/lstm/xprize_predictor.py:6]: ok

File: validation/scenario_generator.py (hunk 3 above)
Comment (ofrancon) [reply validation/scenario_generator.py:55]: The copy is actually only needed when not filtering by country (the filtering will result in a copy). But I think it's clearer to explicitly make a copy first. 👍

**Review by ofrancon — CHANGES_REQUESTED** [review]
The Pandas warning from `pd.read_csv` has been fixed in PR #93 in a more specific way - I think you should remove the `low_memory=False` line from `xprize_predictor.py`. 
The other 2 fixes can be merged, thanks.

**Comment by dsargent** [comment]
I will also add a fix to this PR to clean the CodeFresh environment before running. Need to look up the step that Dan used in ENN.

**Comment by dsargent** [comment]
Added clean-up steps for CodeFresh before and after build. We should probably do this in _all_ builds. It makes things slower but more predictable

**Review by ofrancon — APPROVED** [review]
Looks good, thanks

**Pull request merged**